fusequery/query/src/datasources/system/numbers_stream.rs

```diff
@@ -123,6 +139,13 @@ impl Stream for NumbersStream {
                         self.schema.clone(),
                         vec![Arc::new(UInt64Array::from(arr_data))],
                     );
```

Comment:
Why not abstract it in `ProgressStream`, we can update the progress while consuming the blocks.

Reply:
Yes, good idea.
This is a experience pr.